warp10/src/main/java/io/warp10/script/functions/WSAUDIT.java

```diff
@@ -0,0 +1,45 @@
+//
+//   Copyright 2023  SenX S.A.S.
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+
+package io.warp10.script.functions;
+
+import io.warp10.script.NamedWarpScriptFunction;
+import io.warp10.script.WarpScriptAuditStatement;
+import io.warp10.script.WarpScriptException;
+import io.warp10.script.WarpScriptStack;
+import io.warp10.script.WarpScriptStackFunction;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class WSAUDIT extends NamedWarpScriptFunction implements WarpScriptStackFunction {
+
+  public WSAUDIT(String name) {
+    super(name);
+  }
+
+  @Override
+  public Object apply(WarpScriptStack stack) throws WarpScriptException {
+
+    List l = new ArrayList();
+    for (WarpScriptAuditStatement st: (List<WarpScriptAuditStatement>) stack.getAttribute(WarpScriptStack.ATTRIBUTE_PARSING_ERRORS)) {
```

Comment:
Iterate over a List<Object> and throw and exception if the list encountered something else than WarpScriptAuditStatement. Or simply ignore elements which are not WarpScriptAuditStatement instances.

Reply:
- Returns an empty list when not is audit mode (null stack attribute).
- Returns only WarpScriptAuditStatement as MAPs.